node/src/actors/chain_manager/handlers.rs

```diff
@@ -399,6 +399,10 @@ impl Handler<AddBlocks> for ChainManager {
                         log::debug!("TargetNotReached: superblock target #{} not reached, requesting more blocks. ({} processed blocks)",
                             sync_target.superblock.checkpoint, num_processed_blocks);
                         self.request_blocks_batch(ctx);
+
+                        // Copy current chain state into previous chain state, and persist it
+                        self.move_chain_state_forward(None);
```

Comment:
`self.move_chain_state_forward(None)` doesn't do anything, I think we can remove this line and revert the option back to a u32. Same with `.take()`. But we need to document that `self.persist_chain_state(None, ctx)` always persists the current chain state.


```suggestion
```